test/EndToEndTests/Services/ODataDefaultService/DefaultWCFSvcDataSource.cs

```diff
@@ -1205,15 +1206,15 @@ protected override IEdmModel CreateModel()
             return DefaultInMemoryModel.CreateODataServiceModel("Microsoft.Test.OData.Services.ODataWCFService");
         }
 
-        protected override void ConfigureContainer(IContainerBuilder builder)
+        protected override void ConfigureContainer(IServiceCollection services)
         {
-            base.ConfigureContainer(builder);
-            builder.AddServicePrototype(new ODataSimplifiedOptions()

```

Comment:
Where is `AddServicePrototype` defined? Can we see what it's doing under the hood?

Reply:
@habbes          
''' c sharp
/// <param name="instance">The service prototype to add.</param>
        /// <returns>The <see cref="IContainerBuilder"/> instance itself.</returns>
        public static IContainerBuilder AddServicePrototype<TService>(
            this IContainerBuilder builder,
            TService instance)
        {
            Debug.Assert(builder != null, "builder != null");

            return builder.AddService(ServiceLifetime.Singleton, sp => new ServicePrototype<TService>(instance));
        }
'''